### ISSUE TITLE
FR: Switch for directing to minified (or other classifier) versions of resources.

### ISSUE THREAD

**Issue by metasim** [webjars/webjars-locator#81]
I'm imaging a feature on `WebJarAssetLocator`, or perhaps `RequireJS`, whereby a classifier tag (e.g. `min`) could be specified and, by convention, if a requested resource has a registered version of it with that tag on it, then it is loaded instead.

For example (pseudo-Scala):
```scala
val loc = new WebJarAssetLocator()
loc.registerOverrideClassifier("min")
val path = loc.getFullPath("angularjs", "angular.js")
assert(path === "/webjars/angularjs/1.4.0/angular.min.js")
```

You might want some sort of flexible heuristic for appending the classifier. So you could do something like this:

```scala
val loc = new WebJarAssetLocator()
loc.registerOverrideClassifier("min")
loc.registerOverrideClassifier("gz")
val path = loc.getFullPath("angularjs", "angular.js")
assert(path === "/webjars/angularjs/1.4.0/angular.min.js.gz")
```
Of course if a resource of that name is not available, it falls back to the standard resolution rules.

**Comment by metasim** [comment]
PS: This is a minimally informed and thought-out idea, so no hard feelings if you close it as "bad idea". :-)

**Comment by jamesward** [comment]
I think it is a great idea!  Just need to figure out the right way handle this.  Will noodle this a bit.

**Comment by metasim** [comment]
Now that I think more about it, probably bad idea to add a stateful method like `registerOverrideClassifier`. Would be better to have a version of `getFullPath` that accepts a `Seq[String]` (or equiv.) with the classifiers to consider.

**Comment by mwanji** [comment]
It might be better to pass in an implementation of an interface, so that
name creation can be customised. For example, some libraries use -min
instead of .min.

Also, consider how partial matches are handled. Eg. min and gz are
requested, but only one is available, or each is available separately, but
not together?

**Comment by metasim** [comment]
@mwanji +1! Some sort of resolver callback?

**Comment by jamesward** [comment]
Maybe an optional param on `getFullPath` like `tryToFindMin` which given `foo.js` would try to locate `foo.min.js` or `foo-min.js` via a regex.

**Comment by mwanji** [comment]
I'm not convinced that a method argument is better than a constructor
argument. A typical use case is to use unminified assets in dev and
minified in production. When would you mix the two?

**Comment by jamesward** [comment]
Yeah, you are right.  And I could make it so that `webjars-play` would look at a config param to decide what to set in the constructor.

**Comment by metasim** [comment]
The idea of a functional approach to selection resonates with me. I assume efficiency is a bit of a concern here, so the function would need to be called selectively. 

Would something like this as an argument to the `WebJarAssetLocator` constructor work?

```scala
def artifactSelector(resourceName: String, candidates: Seq[String]): String = { ... }
val loc = new WebJarAssetLocator(artifactSelector)
...
```
(or whatever the closest Java equivalent would be).

**Comment by jvmlet** [comment]
+1 for this feature request
